### PR TITLE
Update script for PDG table

### LINKED ISSUE #13323: Script to update PDG table
### Feature description

It would be good to have a script to update the [PDG table in ROOT](https://github.com/root-project/root/blob/master/etc/pdg_table.txt) from various sources. As a first step, the recent [mcdata from the Particle Data Group](https://pdg.lbl.gov/2023/mcdata/mass_width_2023.txt) could be parsed. As a second step, corresponding [web/database APIs](https://pdg.lbl.gov/2023/html/computer_read.html) could be accessed. 

_Originally suggested by @vkucera in https://github.com/root-project/root/issues/12968#issuecomment-1592966409_

### Alternatives considered

_No response_

### Additional context

This was conceived as a traineeship task by @Axel-Naumann for @DenizCiger.

### PR DESCRIPTION
**Pull request by DenizCiger**
# This Pull request:
Adds a python script to automatically update the pdg_table.txt from the [Particle Data Group](https://pdg.lbl.gov/2023/mcdata/mass_width_2023.txt)
## Changes or fixes:
none

## Checklist:

- [X] tested changes locally
- [ ] updated the docs (if necessary)

This PR fixes #13323

### CODE REVIEW

**Comment by bernhardmgruber** [comment]
@vkucera, since you initially suggested such a script in [a comment](https://github.com/root-project/root/pull/12968#issuecomment-1592966409), you may also want to have a look on this PR. Furthermore, we also tried the [PDG's python API](https://pdgapi.lbl.gov/doc/pythonapi.html), but it can only give us the particle mass, not the width. I contacted them by email to ask how we can get the width. You wouldn't know that by any chance? :)

File: etc/pdg_table_update.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        width = float(elements[8])
+            
+        if trueId in massWidthData:
+            if massWidthData[trueId]['Mass']:
+                mass = float(massWidthData[trueId]['Mass'])
+            if massWidthData[trueId]['Width']:
+                width = float(massWidthData[trueId]['Width'])
+                
+        # Formatting the output line with the updated mass and width
+        return f"{elements[0]:>5} {name:<16} {trueId:>6} {elements[3]:>2} {elements[4]:>3} {elements[5]:<10} {elements[6]:>2} {mass:<11e} {width:<11e} {elements[9]:>3} {elements[10]:>2} {elements[11]:>3} {elements[12]:>2} {elements[13]:>4} {elements[14]:>3}\n"
```
Comment (Axel-Naumann) [inline etc/pdg_table_update.py:53]: This seems to simply print the old data, not the updated one? It seems it simply reproduces the original values, with slightly changed format.

You can see that e.g. `eta_c` has the old mass value (2.980300e+00) not the one from `mass_width_2023.txt` which would be 2.9839E+00, i.e. old is 2.9803 but should now be 2.9839.

Or did I misunderstand the code?

**Review by Axel-Naumann — CHANGES_REQUESTED** [review]
Awesome contribution, thank you so much! I *think* the code has a little bug - if you find the time, could you have a look at my comment, please?

Either way: great work, thank you! If you don't find the time anymore let me know and I will try to fix it, should be simple :-)

**Comment by vkucera** [comment]
> 

Hi @bernhardmgruber , thanks a lot for materialising this idea and for keeping me in the loop. Unfortunately, I cannot help much with the API as I never used it but I hope that the someone from PDG will help.

**Comment by bernhardmgruber** [comment]
> Hi @DenizCiger, @bernhardmgruber , thanks a lot for materialising this idea and for keeping me in the loop. Unfortunately, I cannot help much with the API as I never used it but I hope that the someone from PDG will help.

We are in tough with the PDG on their python API, so we would no longer need to parse the text file. However, they currently only offer access to the masses. widths are not supported yet. so for now, we need to rely on the text file.

File: etc/pdg_table_update.py (hunk 1 above)
Comment (DenizCiger) [reply etc/pdg_table_update.py:53]: Thanks for the feedback! I'll fix the bug right away!

**Comment by bernhardmgruber** [comment]
@DenizCiger I think your last commit now produces the correct values.  Great!
@Axel-Naumann would you like to have all changes squashed into one commit? We previously separated the script from the update to the txt file. The latest commit changes both now, which is not elegant.

**Comment by Axel-Naumann** [comment]
@bernhardmgruber sure, I'll just squash when merging, let's not spend time on this.

**Review by Axel-Naumann — APPROVED** [review]
Looks great, thanks! I'll merge once (some of) the tests succeeded.

**Comment by bernhardmgruber** [comment]
Alright, thx a lot!

**Comment by Axel-Naumann** [comment]
Congratulations, @DenizCiger and thanks again for your very nice contribution!

**Comment by smuzaffar** [comment]
@DenizCiger @Axel-Naumann , this added dependency on `#!/usr/bin/python` which on rhel8/9 is not available. Can this be changed to `either `#!/usr/bin/python3` or `#!/usr/bin/env python` ?

**Comment by bernhardmgruber** [comment]
@smuzaffar we just put into the shebang what worked for us. I am not knowledgeable on how to find python cross-platform, so I will hand this over to you or @Axel-Naumann to provide a corresponding PR.

**Pull request merged**